include/extractor/geojson_debug_policies.hpp

```diff
@@ -37,6 +37,21 @@ struct IntersectionPrinter
     const extractor::guidance::CoordinateExtractor &coordinate_extractor;
 };
 
+struct IntersectionPrinter : public IntersectionShapePrinter
+{
+    typedef IntersectionShapePrinter Base;
+    IntersectionPrinter(const util::NodeBasedDynamicGraph &node_based_graph,
+                        const std::vector<extractor::QueryNode> &node_coordinates,
+                        const extractor::guidance::CoordinateExtractor &coordinate_extractor);
+
+    // renders the used coordinate locations for all entries/as well as the resulting
+    // intersection-classification
+    util::json::Array operator()(const NodeID intersection_node,
+                                 const extractor::guidance::Intersection &intersection,
+                                 const boost::optional<util::json::Object> &node_style = {},
+                                 const boost::optional<util::json::Object> &way_style = {}) const;
+};
```

Comment:
I tried to not make it a template to reduce the number of items that we define in headers. Right now the derived class transforms what it is given into an intersectionShape and calls its base class operator. There is no code duplication for rendering, but I can transform it into a template just as well.

Whats your opinion here? I don't feel strongly one way or the other.